script/benchmarker.rb

```diff
@@ -4,14 +4,25 @@
 num_children = 100
 
 # Pop off the indexing_persister  so we don't test it
+Valkyrie.logger.level = :error
 Valkyrie::Adapter.adapters.to_a.each do |adapter_name, adapter|
   Benchmark.bm do |bench|
     parent = Book.new
     children = nil
     bench.report("#{adapter_name} create #{num_children} children") do
-      children = Array.new(num_children) do |_page_num|
-        p = Page.new
-        adapter.persister.save(model: p)
+      children = []
+      if adapter.persister.respond_to?(:buffer_into_index)
+        adapter.persister.buffer_into_index do |persist|
+          children = Array.new(num_children) do |_page_num|
+            p = Page.new
+            persist.save(model: p)
+          end
+        end
+      else
+        children = Array.new(num_children) do |_page_num|
+          p = Page.new
+          adapter.persister.save(model: p)
+        end
```

Comment:
To be clear, in that pattern, a commit for a non-buffering persister would be a no-op. This is a little different than transactions, though. Just thinking through the options of semantics vs. type checking, and looking for a tidy interface with few surprises.

I think the block is the right choice for now and maybe forever -- but maybe the notion of "do stuff in a block" is a top level thing with a generic name. I guess some of the use cases would be about controlling sequencing (like doing stuff before or after the chain of command).

Not for now, either way.